File: src/types/collectionPoint.ts
export interface CollectionPoint {
  id: string;
  name: string;
  address: string;
  latitude: number;
  longitude: number;
  capacity: number;
  currentCount: number;
  fillRate: number; // 0 - 100
  lastCollectedAt?: string;
  estimatedNextAvailableAt?: string;
  distanceMeters?: number;
}

File: gcycle-backend/app/schemas/collection_point.py
from pydantic import BaseModel

from app.core.constants import CollectionPointStatus


class CollectionPointOut(BaseModel):
    id: str
    name: str
    address: str
    lat: float
    lng: float
    capacity: int
    current_count: int
    fill_rate: float
    status: CollectionPointStatus

    class Config:
        from_attributes = True
